scenario_player/utils/legacy.py

```diff
@@ -197,35 +197,36 @@ def _set_timeout(self, timeout=None):
             self._endtime = time.time() + timeout
 
 
-def wait_for_txs(
-    client_or_web3: Union[Web3, JSONRPCClient], txhashes: Set[TransactionHash], timeout: int = 360
-):
-    if isinstance(client_or_web3, Web3):
-        web3 = client_or_web3
-    else:
-        web3 = client_or_web3.web3
+def wait_for_txs(client: JSONRPCClient, txhashes: Set[TransactionHash], timeout: int = 360):
+
     start = time.monotonic()
     outstanding = False
-    txhashes = set(txhashes)
-    while txhashes and time.monotonic() - start < timeout:
+    txhashes_bytes = []
+
+    ## for some reason txhashes is a set of str, not bytes
+    for txhash in txhashes:
+        if isinstance(txhash, str):
+            txhashes_bytes.append(bytes.fromhex(remove_0x_prefix(txhash)))
+
+    while txhashes_bytes and time.monotonic() - start < timeout:
         remaining_timeout = timeout - (time.monotonic() - start)
-        if outstanding != len(txhashes) or int(remaining_timeout) % 10 == 0:
-            outstanding = len(txhashes)
+        if outstanding != len(txhashes_bytes) or int(remaining_timeout) % 10 == 0:
+            outstanding = len(txhashes_bytes)
             log.debug(
                 "Waiting for tx confirmations",
                 outstanding=outstanding,
                 timeout_remaining=int(remaining_timeout),
             )
-        for txhash in txhashes.copy():
-            tx = web3.eth.getTransactionReceipt(txhash)
+        for txhash in txhashes_bytes.copy():
+            tx = client.poll(txhash)
```

Comment:
This probably fixes the issue, but is suboptimal since it blocks on each tx until they are mined.

Instead I'd suggest to add the logic from poll in here.
Basically that's only adding a check that the block number returned from `eth.getTransactionReceipt` is `current_block` + `confirmation_blocks`.